smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/HttpPayloadValidator.java

```diff
@@ -95,13 +100,32 @@ private Optional<ValidationEvent> validatePayload(
                 .map(Map.Entry::getKey)
                 .collect(Collectors.toSet());
 
+        List<ValidationEvent> events = new ArrayList<>();
+
         if (!unbound.isEmpty()) {
-            return Optional.of(error(inputOrError, String.format(
+            events.add(error(inputOrError, String.format(
                     "A member of this structure is marked with the `httpPayload` trait, but the following "
                     + "structure members are not explicitly bound to the HTTP message: %s",
                     ValidationUtils.tickedList(unbound))));
         }
 
-        return Optional.empty();
+        Optional<HttpBinding> payloadBinding = resolved.values().stream()
+                .filter(binding -> binding.getLocation().equals(HttpBinding.Location.PAYLOAD))
+                .findFirst();
+        if (payloadBinding.isPresent() && needsRequiredOrDefault(model, payloadBinding.get())) {
+            events.add(error(payloadBinding.get().getMember(),
+                    "Members marked with the `httpPayload` trait and `streaming` trait that target strings or "
+                    + "blobs MUST also be marked with the `required` or `default` trait."));
+        }
+
+        return events;
+    }
+
+    private boolean needsRequiredOrDefault(Model model, HttpBinding payloadBinding) {
+        MemberShape member = payloadBinding.getMember();
+        Shape target = model.expectShape(member.getTarget());
+        return (target.isStringShape() || target.isBlobShape())
```

Comment:
`@streaming` can only be on the `target` and can only target `blob` for defaulting purposes, simplifying this a little.